src/net/socket.rs

```diff
@@ -60,34 +59,62 @@ impl Socket {
     /// endpoint by looking to see if they are still sending packets or not
     ///
     /// This function allows you to configure laminar with the passed configuration.
-    pub fn bind_with_config<A: ToSocketAddrs>(
-        addresses: A,
-        config: Config,
-    ) -> Result<(Self, Sender<Packet>, Receiver<SocketEvent>)> {
+    pub fn bind_with_config<A: ToSocketAddrs>(addresses: A, config: Config) -> Result<Self> {
         let socket = UdpSocket::bind(addresses)?;
         Self::bind_internal(socket, config)
     }
 
-    fn bind_internal(
-        socket: UdpSocket,
-        config: Config,
-    ) -> Result<(Self, Sender<Packet>, Receiver<SocketEvent>)> {
+    fn bind_internal(socket: UdpSocket, config: Config) -> Result<Self> {
         socket.set_nonblocking(true)?;
         let (event_sender, event_receiver) = unbounded();
         let (packet_sender, packet_receiver) = unbounded();
-        Ok((
-            Socket {
-                recv_buffer: vec![0; config.receive_buffer_max_size],
-                socket,
-                config,
-                connections: ActiveConnections::new(),
-                link_conditioner: None,
-                event_sender,
-                packet_receiver,
-            },
-            packet_sender,
-            event_receiver,
-        ))
+        Ok(Socket {
+            recv_buffer: vec![0; config.receive_buffer_max_size],
+            socket,
+            config,
+            connections: ActiveConnections::new(),
+            link_conditioner: None,
+            event_sender,
+            packet_receiver,
+
+            sender: packet_sender,
+            receiver: event_receiver,
+        })
+    }
+
+    /// Get a sender queue
```

Comment:
Technically these are handles to the internal queuing mechanism.

Reply:
What do you propose?

Reply:
```
/// Returns a handle to the packet sender which provides a thread-safe way to enqueue packets to be processed. This should be used when the socket is busy running its polling loop in a separate thread. 
pub fn get_packet_sender(&mut self) -> Sender<Packet>

/// Returns a handle to the event receiver which provides a thread-safe way to retrieve events from the socket. This should be used when the socket is busy running its polling loop in a separate thread.
pub fn get_event_receiver(&mut self) -> Receiver<SocketEvent>
```

Reply:
Ok done